Add per-player block reach

Camotoy opened the pull request:
This commit implements the outline I presented in #4908. Most importantly, I have confirmed that all vanilla values remain correct without modification from the API, and when resetting block range back to their vanilla values. Please do let me know if there's something I can improve (especially documentation).

I'm also willing to release these patches under MIT, if there is a process for that. 

Fixes #4908

Camotoy commented on Spigot-Server-Patches/0619-Add-per-player-block-reach.patch:
```diff
@@ -0,0 +1,129 @@
++
++    @Override
++    public void setSurvivalBlockRange(int range) {
++        range = Math.max(6, range); // Ensure range can't be set lower than vanilla defaults, or else block ghosting will occur for vanilla clients
```
After some discussion with someone else - should this be replaced with a `Preconditions` check?

zachbr replied:
Yes. Existing API design prefers throwing and letting plugins deal with the consequences over silently adjusting the value.
Though like everything, there are exceptions.

Camotoy replied:
Addressed.

Camotoy committed:
Add per-player block reach

Plugins are able to specify the server-accepted block range of a specific player.
